### ISSUE TITLE
Missing gfx942 KDB file in ROCm 6.2 release for RHEL

### ISSUE THREAD

**Issue by groenenboomj** [ROCmSoftwarePlatform/MIOpen#3242]
https://repo.radeon.com/rocm/rhel9/6.2/main/

Missing the KDB file and a bit more in the ROCm 6.2 release for RHEL targets for gfx942.  This is causing packing and building issues.

Was it missed in this release?

**Comment by groenenboomj** [comment]
Actually it looks like it's missing from all releases.

**Comment by groenenboomj** [comment]
Specifically this causes an error when building ROCm.  The MIOpen build looks for the KDB files and issues an error when not found.

**Comment by junliume** [comment]
@groenenboomj could you please share the log on the "error when building ROCm"?

**Comment by groenenboomj** [comment]
I am getting a reproducer together but we may be able to short circuit this with one question. Are KDB files no longer being released for MI300 targets?

**Comment by trixirt** [comment]
@groenenboomj could you take a look at Fedora's spec file https://src.fedoraproject.org/rpms/miopen/blob/rawhide/f/miopen.spec ?  This should give you a starting point on how to build.

**Comment by prarit** [comment]
@junliume @trixirt the issue does not appear to be with a MIOpen build, but rather an issue when building other packages which rely on the MIOpen packages.

tl;dr: it looks like AMD is missing MIOpen gfx942 kdb packages which are required to build torch for AMD ROCm.

https://repo.radeon.com/rocm/rhel9/6.2/main/ << this repo is missing MIOpen gfx942 kdb packages

We use a container to setup our build environment and we do

 dnf install -y --nodocs \
         --enablerepo=amdgpu-${ROCM_VERSION} --enablerepo=ROCm-${ROCM_VERSION} \
         rocm-dev \
         rocm-libs \
         miopen-hip-gfx90akdb \
         miopen-hip-gfx942kdb

which works for 6.1.2, but fails for 6.2.  Back to @groenenboomj question: " Are KDB files no longer being released for MI300 targets?"

**Comment by groenenboomj** [comment]
> @groenenboomj could you take a look at Fedora's spec file https://src.fedoraproject.org/rpms/miopen/blob/rawhide/f/miopen.spec ?  This should give you a starting point on how to build.

Agreed that this should work but as @prarit said, not having KDB files can cause us packaging and performance issues unless you're telling us we no longer need them.

@junliume @trixirt Does this issue need moved to someone else for the KDB question?

**Comment by cderb** [comment]
gfx942.kdb was removed in #3026. This was to avoid conflicts with GPU target flags. The kdb could be re-added later, but currently there are no plans to. The kdbs are built up to support explicitly tuned entries and MIOpen is shifting towards using a heuristic model for inference instead.

**Comment by junliume** [comment]
@prarit missing of the kdb file should not be causing any build failures, could you elaborate what the packaging issue are?

**Comment by groenenboomj** [comment]
> gfx942.kdb was removed in #3026. This was to avoid conflicts with GPU target flags. The kdb could be re-added later, but currently there are no plans to. The kdbs are built up to support explicitly tuned entries and MIOpen is shifting towards using a heuristic model for inference instead.

Okay, then we have a non issue. We can close this.

**Comment by atamazov** [comment]
@cderb 

> This was to avoid conflicts with GPU target flags.

Can you please look at https://github.com/ROCm/MIOpen/pull/3026#issuecomment-2353214705, thanks,

**Comment by groenenboomj** [comment]
We will revisit this topic if KDB files are reintroduced. Thanks for the clarification. I'm excited to see if this simplification doesn't have performance impacts but simpler is better.
